### PR TITLE
Reduce RPC cost of oscar by removing unnecessary tasks

### PR DESCRIPTION
**Pull request by wjsi**
## What do these changes do?

Reduce RPC cost of oscar by removing unnecessary tasks. Now Oscar can handle 4500 calls pser sec instead of 2500 before optimization.

## Related issue number

Fixes #xxxx

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/oscar/backends/context.py [hunk 1]
```diff
@@ -70,8 +70,17 @@ def _process_result_message(message: Union[ResultMessage, ErrorMessage]):
...
+        def future_setter(_):
+            try:
+                fut.set_result(True)
+            except asyncio.InvalidStateError:
+                pass
+
+        future.add_done_callback(future_setter)
         try:
-            await asyncio.wait([future])
+            await fut
```
Comment (hekaisheng) [inline mars/oscar/backends/context.py:83]: Seems `asyncio.shield` looks more natural, could you add some explanations about it?

File: mars/oscar/core.pyx [hunk 2]
```diff
@@ -228,20 +228,30 @@ cdef class _BaseActor:
...
-                    tasks.append(value)
+                    value = res_item
+                    coros.append(value)
+                    coro_poses.append(idx)
                 else:
                     value = res_item
                 values.append(value)
 
-            if len(tasks) > 0:
+            if len(coros) == 1:
```
Comment (hekaisheng) [inline mars/oscar/core.pyx:247]: Please add some comments here.

**Review by hekaisheng — COMMENTED** [review]
LGTM, some comments may be added.

**Review by hekaisheng — APPROVED** [review]
LGTM

File: mars/oscar/backends/context.py [hunk 3]
```diff
@@ -71,13 +71,15 @@ def _process_result_message(message: Union[ResultMessage, ErrorMessage]):
...
-            await asyncio.wait([future])
+            await asyncio.shield(future)
         except asyncio.CancelledError:
             try:
                 await self.cancel(address, message.message_id)
             except CannotCancelTask:
                 # cancel failed, already finished
                 raise asyncio.CancelledError
+        except:  # noqa: E722  # nosec  # pylint: disable=bare-except
+            pass
```
Comment (chaokunyang) [inline mars/oscar/backends/context.py:82]: Why ignore the exception？

Comment (wjsi) [reply mars/oscar/backends/context.py:82]: > 

To make the result consistent with `async.wait` which does not raises.

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**